scipy/cluster/tests/test_hierarchy.py

```diff
@@ -842,26 +749,24 @@ def test_is_monotonic_Q_linkage(self, xp):
         assert is_monotonic(Z)
 
 
+@skip_if_array_api(cpu_only=True)
 class TestMaxDists:
 
-    @skip_if_array_api_gpu
-    @array_api_compatible
+    @skip_if_array_api(cpu_only=True)
     def test_maxdists_empty_linkage(self, xp):
         # Tests maxdists(Z) on empty linkage. Expecting exception.
         Z = xp.zeros((0, 4), dtype=xp.float64)
         assert_raises(ValueError, maxdists, Z)
 
-    @skip_if_array_api_gpu
-    @array_api_compatible
+    @skip_if_array_api(cpu_only=True)
     def test_maxdists_one_cluster_linkage(self, xp):
         # Tests maxdists(Z) on linkage with one cluster.
         Z = xp.asarray([[0, 1, 0.3, 4]], dtype=xp.float64)
         MD = maxdists(Z)
         expectedMD = calculate_maximum_distances(Z, xp)
         xp_assert_close(MD, expectedMD, atol=1e-15)
 
-    @skip_if_array_api_gpu
-    @array_api_compatible
+    @skip_if_array_api(cpu_only=True)
```

Comment:
Here and above -- there's a class-level skip doing this now, right?

Reply:
Correct, I'll change this.